overlord/overlord.go

```diff
@@ -167,6 +168,36 @@ func (o *Overlord) Stop() error {
 	return err1
 }
 
+// Settle runs first a state engine Ensure and then wait for activities to settle.
+// That's done by waiting for all managers activities to settle while
+// making sure no immediate further Ensure is scheduled. Chiefly for tests.
+// Cannot be used in conjuction with Run.
+func (o *Overlord) Settle() {
+	o.ensureLock.Lock()
+	if o.ensureTimer != nil {
+		panic("cannot use Settle with an ensure loop (Overlord.Run)")
+	}
+	o.ensureTimer = time.NewTimer(0)
+	defer func() {
+		o.ensureLock.Lock()
+		defer o.ensureLock.Unlock()
+		o.ensureTimer.Stop()
+		o.ensureTimer = nil
+	}()
+	o.ensureLock.Unlock()
```

Comment:
For correctness, we should probably avoid the panic inside a mutex region doing something like
```
lock
hadTimer := o.ensureTimer != nil
...
unlock
defer
if hadTimer { panic }
```